### PR TITLE
Setup buildSrc repository with shared build logic

### PR DESCRIPTION
**Pull request by jmfayard**
Follow-up from this pull-request
https://github.com/inorichi/tachiyomi/pull/1672

See TODOs in ./build.gradle

Also update Gradle and the Android Studio Gradle plugin to the latest stable version

### CODE REVIEW

File: buildSrc/src/main/kotlin/Libs.kt [hunk 1]
```diff
@@ -0,0 +1,220 @@
+import kotlin.String
```
Comment (inorichi) [inline buildSrc/src/main/kotlin/Libs.kt]: Is this import really needed?

File: buildSrc/src/main/kotlin/Libs.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+/**
+ * Generated by [gradle-kotlin-dsl-libs](https://github.com/jmfayard/gradle-kotlin-dsl-libs)
+ *
+ * Run again
+ *   `$ ./gradlew syncLibs`
+ * to update this file */
+object Libs {
+    /**
+     * [appcompat website](http://developer.android.com/tools/extras/support-library.html) */
+    const val appcompat: String = "androidx.appcompat:appcompat:" + Versions.appcompat
```
Comment (inorichi) [inline buildSrc/src/main/kotlin/Libs.kt]: `: String` shouldn't be needed.

**Comment by inorichi** [comment]
I didn't ask last time... Does autocompletion work on the default groovy scripts?

File: buildSrc/src/main/kotlin/Libs.kt (hunk 1 above)
Comment (jmfayard) [reply buildSrc/src/main/kotlin/Libs.kt]: No, looks like a small issue in https://github.com/kotlintest/kotlintest

**Pull request merged**